src/Form/organisms/AvatarUploader.tsx

```diff
@@ -0,0 +1,156 @@
+  align-items: center;
+  justify-content: center;
+  border-radius: 5px;
+  background-color: hsl(0, 0%, 90%);
+`; 
+const PreviewImage = styled.img`
+  ${PhotoContainerStyle}
+`;
+
+const PlaceholderText = styled.div`
+  color: hsl(207, 5%, 57%);
+  font-size: 14px;
+`;
+
+const NoPhoto = styled.div`
+  ${PhotoContainerStyle}
+  padding-top: 15px;
+  ${PlaceholderText} {
+    margin-top: 18px;
+  }
+`;
+
+/**
+ * TODO: Free memory for  URL.createObjectURL(); 
+ * https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL
+ * 
+*/
+
+const StyledInputFileButton = styled(InputFileButton)`
+  width: 100%;
+`;
+
+interface IAvatar {
+  title?: string
+  innerText?: string
+  buttonText?: string
+  buttonTextReplace?: string
+  onAvatarReady?: (imgFile: File) => void
+  onError?: (msg: string) => void
+}
+
+const AvatarUploader : React.FC<IAvatar> = ({
+  title = 'Photograph',
+  innerText = 'Drop Photo',
+  buttonText = 'Select File',
+  buttonTextReplace = 'Replace Photo',
+  onAvatarReady,
+  onError = () => {},
+}) => {
+  const [avatarImg, setAvatarImg] = useState('');
+  const [isCropOpen, setIsCropOpen] = useState(false);
+
+  const handleCrop = useCallback(async (newFileUrl: string) => {
+    setAvatarImg(newFileUrl);
+    let newFile = await fetch(
+      newFileUrl).then(r => r.blob()).then(blobFile => new File([blobFile], "newAvatar", { type: "image/png" })
+      );
+    if(onAvatarReady) {
+      onAvatarReady(newFile);
+    }
+    setIsCropOpen(false);
+  },[onAvatarReady]);
+
+  const handleCropClose = useCallback(() => {
+    setIsCropOpen(false);
+  },[]);
+
+  const handleFileUpload  = useCallback((newFiles: FileList) => {
+
+    if(newFiles.length === 1) {
+      if(!isValidImage(newFiles[0])){ 
+        onError('Please upload only jpeg and png file');
+        return;
+      }
+      const prevImg = URL.createObjectURL(newFiles[0]);
+
+      setAvatarImg(prevImg);
+      setIsCropOpen(true);
+    } else {
+      onError('Drop only one file');
+    }
+  },[onError]);
+
+  return(
+    <Container>
+      <Label labelText={title} htmlFor='avatar-upload' />
+      <PreviewImageGroup>
+        {avatarImg
+          ? <PreviewImage src={avatarImg} alt='avatar image' />
+          : (
+            <NoPhoto>
+              <AvatarPlaceholder />
+              <PlaceholderText>{innerText}</PlaceholderText>
+            </NoPhoto>
+            )}
+        <DropArea height={PHOTO_HEGHT} dropCallback={handleFileUpload} />
+      </PreviewImageGroup>
+      <StyledInputFileButton
+        id='avatar-upload'
+        text={avatarImg ? buttonTextReplace : buttonText}
+        buttonSize='small'
+        accept='image/*'
+        callback={handleFileUpload}
+      />
+      {isCropOpen
+        ? <CropTool
+            imgUrl={avatarImg}
+            onCrop={handleCrop}
+            onClose={handleCropClose}
+            canvasHeight={490}
+            canvasWidth={460}
+            cropHeight={360}
+            cropWidth={360}
```

Comment:
Are these just defaults? Wasn't expecting to see hard coded values here for `CropTool` so it seems like these could be a little problematic when you have unexpected aspect ratios. Could this cause any issues?

Reply:
The reason behind the Croptool getting this values look "hard coded" is that the parent of the croptool is the one that controls the size of the canvas and the size of the selection. (although if the size of the selection is is bigger it will be ajusted to the canvas).  

From the examples I reviewed the tools allow the user of the UI library to provide this values, although sometimes allow to receive an aspect radio instead of those and the UI plugin would calculate the canvas size for them. Maybe we can have a call to better understand the expectations.